spec/requests/api/graphql/queries/consignments_spec.rb

```diff
@@ -0,0 +1,217 @@
+
+        error_message = body['errors'][0]['message']
+        expect(error_message).to eq "Can't find partner."
+      end
+    end
+
+    context 'with a request from a regular user' do
+      let(:token) do
+        payload = { aud: 'gravity', sub: 'userid', roles: 'user' }
+        JWT.encode(payload, Convection.config.jwt_secret)
+      end
+
+      it 'returns an error for that request' do
+        post '/api/graphql', params: { query: query }, headers: headers
+
+        expect(response.status).to eq 200
+        body = JSON.parse(response.body)
+
+        consignments_response = body['data']['consignments']
+        expect(consignments_response).to eq nil
+
+        error_message = body['errors'][0]['message']
+
+        expect(error_message).to eq "Can't find partner."
+      end
+    end
+  end
+
+  describe 'valid requests' do
+    it 'returns the expected graphql response' do
+      post '/api/graphql', params: { query: query }, headers: headers
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+
+      expect(body['data']).to eq(
+        {
+          consignments: {
+            totalCount: 4,
+            edges: [
+              { node: { currency: 'USD', state: 'SOLD' } },
+              { node: { currency: 'USD', state: 'BOUGHT_IN' } },
+              { node: { currency: 'USD', state: 'SOLD' } },
+              { node: { currency: 'USD', state: 'BOUGHT_IN' } }
+            ]
+          }
+        }.deep_stringify_keys
+      )
+    end
+    it 'returns only sold or bought consignments' do
+      post '/api/graphql', params: { query: query }, headers: headers
+
+      expect(response.status).to eq 200
+      body = JSON.parse(response.body)
+
+      total_count = body.dig('data', 'consignments', 'totalCount')
+      state1 = body.dig('data', 'consignments', 'edges', 0, 'node', 'state')
+      state2 = body.dig('data', 'consignments', 'edges', 1, 'node', 'state')
+      state3 = body.dig('data', 'consignments', 'edges', 2, 'node', 'state')
+      state4 = body.dig('data', 'consignments', 'edges', 3, 'node', 'state')
+
+      expect(total_count).to eq 4
+      expect(state1).to eq 'SOLD'
+      expect(state2).to eq 'BOUGHT_IN'
+      expect(state3).to eq 'SOLD'
+      expect(state4).to eq 'BOUGHT_IN'
+    end
+
+    describe 'sorting' do
+      let(:query) do
+        <<-GRAPHQL
+        query {
+          consignments(#{query_inputs}) {
+             totalCount
+              edges {
+                node {
+                  currency
+                  id
+                  internalID
+                  saleDate
+                  saleName
+                  salePriceCents
+                  state
+                  submissionID
+                  submission {
+                    artistId
+                    category
+                    title
+                  }
+                }
+              }
+            }
+          }
+        GRAPHQL
+      end
```

Comment:
**suggestion [late-feedback]:** remove unused fields

I love this set of sorting specs, well done!! I just have a minor suggestion that we only ask for the fields in the query that will be used in the tests:

```ruby
      let(:query) do
        <<-GRAPHQL
        query {
          consignments(#{query_inputs}) {
              edges {
                node {
                  id
                }
              }
            }
          }
        GRAPHQL
      end
```

This way it's more clear to our future selves that these tests are strictly about sorting.

*format via [Conventional Comments](https://conventionalcomments.org/)*